### PR TITLE
feat(stack): add stack component package

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Stack component package
- [x] Stack stories for Storybook
- [x] Stack tests

### CODE REVIEW

File: packages/paste-core/layout/stack/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      }
+      if (value === 'vertical') {
+        marginRightValue = 'space0';
+        marginBottomValue = spacing;
+      }
+
+      // @ts-ignore
+      horizontalMarginArray[i] = marginRightValue;
+      // @ts-ignore
+      verticalMarginArray[i] = marginBottomValue;
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/src/index.tsx:69]: I believe you might be able to simplify this to something like

```
horizontalMarginArray[i] = (value=== 'horizontal') ? spacing : 'space0'

verticalMarginArray[i] = (value=== 'horizontal') ? 'space0' : spacing;
```

Comment (SiTaggart) [reply packages/paste-core/layout/stack/src/index.tsx:69]: In fact could you further simply this entire thing to 
```
const styles: MarginProps = {}

if (Array.isArray(orientation)) {
  const marginRight: Space = [];
  const marginBottom: Space = [];

  orientation.forEach((value, i) => {
    marginRight[i] = (value=== 'horizontal') ? spacing : 'space0'
    marginBottom[i] = (value=== 'horizontal') ? 'space0' : spacing;
  });

  styles = {
    marginRight,
    marginBottom
  }
}

  if (orientation === 'horizontal') {
    styles.marginRight = spacing;
  }

  if (orientation === 'vertical') {
    styles.marginBottom = spacing;
  }

  return styles;
```

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        <Card>Card two</Card>
+      </Stack>
+    </Theme.Provider>
+  );
+};
+
+describe('Stack', () => {
+  it('should render a vertical stack', () => {
+    const {asFragment} = testRender(<MockStack />);
+    expect(asFragment()).toMatchSnapshot();
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/__tests__/stack.test.tsx:34]: This is a much nicer way to do snapshots

File: packages/paste-core/layout/stack/src/index.tsx (hunk 1 above)
Comment (richbachman) [reply packages/paste-core/layout/stack/src/index.tsx:69]: That is much simpler. Just needed to change `const styles...` to `let styles...`. Thanks.

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import {render as testRender} from '@testing-library/react';
+import {Card} from '@twilio-paste/card';
+import {Theme} from '@twilio-paste/theme';
+import {getStackDisplay, Stack} from '../src';
+
+describe('Stack Unit Tests', () => {
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/__tests__/stack.test.tsx:7]: Would be worth having tests for `getStackChildMargins` and `getStackStyles` too, I suspect

**Pull request merged**